### PR TITLE
Add CUDA support for torch.ormqr

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54725 Added cuBLAS path for torch.linalg.lstsq
* #57317 Add cuSOLVER path for torch.linalg.lstsq
* **#57316 Add CUDA support for torch.ormqr**
* #57315 Port CPU torch.ormqr to ATen

CUDA support is implemented using cuSOLVER.

Differential Revision: [D28242071](https://our.internmc.facebook.com/intern/diff/D28242071)

### CODE REVIEW

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cu [hunk 1]
```diff
@@ -1994,9 +1994,21 @@ Tensor& orgqr_kernel_impl(Tensor& result, const Tensor& tau) {
...
       "PyTorch with cuSOLVER. Please use PyTorch built with cuSOLVER support.");
   #endif
-  }
+}
+
+REGISTER_DISPATCH(orgqr_stub, &orgqr_kernel_impl);
+
+void ormqr_kernel(const Tensor& input, const Tensor& tau, const Tensor& other, bool left, bool transpose) {
+#if defined(USE_CUSOLVER)
+  return ormqr_cusolver(input, tau, other, left, transpose);
```
Comment (xwang233) [inline aten/src/ATen/native/cuda/BatchLinearAlgebra.cu:2003]: Is this return within a void function correct?

**Review by xwang233 — APPROVED** [review]
LGTM. Thanks for the work!

File: aten/src/ATen/native/cuda/BatchLinearAlgebra.cu (hunk 1 above)
Comment (IvanYashchuk) [reply aten/src/ATen/native/cuda/BatchLinearAlgebra.cu:2003]: It is a valid code, but you're right, it shouldn't be here.

**Review by mruberry — APPROVED** [review]
Nice work @IvanYashchuk, thanks for reviewing, @xwang233!

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28242071).

**Comment by mruberry** [comment]
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28242071).

**Pull request closed without merging**